frontend/src/backend/socket.ts

```diff
@@ -0,0 +1,22 @@
+import { getPageContext } from '../page-context';
+import * as io from 'socket.io-client';
+
+type SocketPromise = Promise<SocketIOClient.Socket>;
+let socket: SocketPromise;
+
+function createSocket(): Promise<SocketIOClient.Socket> {
+   return new Promise((resolve) => {
+      const socket = io.connect('/');
+
+      socket.on('connect', function() {
+         getPageContext().then((details) => {
+            socket.emit('authenticate', details.socketToken);
+         });
+      });
+      resolve(socket);
```

Comment:
This looks to me like it always resolves immediately, without waiting for anything to happen on the socket. Is that intentional? And if so, why the `Promise`?

Reply:
Yes it's intentional. Callers merely subscribe to more events.
This was the commit: 73395306ed5b943ab8682e468102d2de56960e3d
It took me a bit to remember what I was envisioning. The short of it is I want the `socket` object to be a singleton (so other parts could also use it to communicate) and a promise was an easy way to ensure that. Though on second thought, most usages just create the socket in the module context and don't bother waiting for a `createSocket()` call.

I'll do that. Thanks.

Reply:
Turns out, I didn't want to create it just in the module because in front-end-only mode, there is no backend and we can a consistent string of errors in the console as it tries to connect.

That said, you were right a promise wasn't needed cause we can return right away. Fixed!